packages/base-manager/src/manager-base.ts

```diff
@@ -330,35 +342,57 @@ export abstract class ManagerBase implements IWidgetManager {
     serialized_state: any = {}
   ): Promise<WidgetModel> {
     const model_id = options.model_id;
-    const model_promise = this.loadClass(
+    const model_promise = this.loadModelClass(
       options.model_name,
       options.model_module,
       options.model_module_version
-    ) as Promise<typeof WidgetModel>;
+    );
     let ModelType: typeof WidgetModel;
+
+    const makeErrorModel = (error: any, msg: string) => {
+      const Cls = widgets.createErrorWidget(error, msg);
+      const widget_model = new Cls({
+        _model_module: options.model_module,
+        _model_name: options.model_name,
+      });
+      widget_model.name = options.model_name;
+      widget_model.module = options.model_module;
```

Comment:
Same question here I guess, do we need to set those parameters?

Reply:
Yes you are correct, these are the relics of old implementation when the error messages are generated inside `ErrorWidget`